src/app/code/community/AvS/FastSimpleImport/Model/Import/Entity/Product.php

```diff
@@ -1612,29 +1619,46 @@ protected function _getUploader()
     }
 
     /**
-     * @param $fileName
-     * @return bool
+     * Returns an object for upload downloadable files
+     *
+     * @return Mage_ImportExport_Model_Import_Uploader
      */
-    protected function moveDownloadableFile($fileName)
+    protected function _getDownloadableUploader()
     {
-        $filePath      = $this->_getUploader()->getTmpDir() . $fileName;
-        $basePath      = Mage::getModel('downloadable/link')->getBasePath();
-        $destDirectory = dirname(Mage::helper('downloadable/file')->getFilePath($basePath, $fileName));
-        // make sure that the destination directory exists!
-        $ioObject = new Varien_Io_File();
-        try {
-            $ioObject->open(array('path' => $destDirectory));
-        } catch (Exception $e) {
-            $ioObject->mkdir($destDirectory, 0777, true);
-            $ioObject->open(array('path' => $destDirectory));
-        }
-        $destFile   = $basePath . DS . $fileName;
-        $sourceFile = realpath($filePath);
-        if ($sourceFile !== false) {
-            return copy($sourceFile, $destFile);
-        } else {
-            return false;
+        if (is_null($this->_downloadableUploader)) {
+            // make sure to pass null as a constructor argument, so that no upload file is set!
+            $this->_downloadableUploader = Mage::getModel('fastsimpleimport/import_uploader_downloadable', null);
+            $this->_downloadableUploader->init();
+
+            $tmpDir  = Mage::getConfig()->getOptions()->getMediaDir() . '/import';
+            $destDir = Mage::getModel('downloadable/link')->getBasePath();
+            if ( ! is_writable($destDir)) {
```

Comment:
Is this really what you want to do? Check if not writeable and then create it? I really have strong opinions against the usage of `@` for error suppression like in this case. Either a correct check or a correct handling of the result of the check you already do would be the cleaner alternative.